oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtIssuerReactiveAuthenticationManagerResolver.java

```diff
@@ -166,7 +167,7 @@ public JwtIssuerReactiveAuthenticationManagerResolver(Collection<String> trusted
 					.filter(this.trustedIssuer)
 					.flatMap(iss ->
```

Comment:
if assume that predicate is stateless (as far as result is cached indefinitely) this could be rewritten to  
```java
@Override
public Mono<ReactiveAuthenticationManager> resolve(String issuer) {
	if (this.trustedIssuer.test(issuer)) {
		return Mono.empty();
	}
	return this.authenticationManagers.computeIfAbsent(issuer, k ->
			Mono.<ReactiveAuthenticationManager>fromCallable(() ->
					new JwtReactiveAuthenticationManager(ReactiveJwtDecoders.fromIssuerLocation(k))
			)
			.subscribeOn(Schedulers.boundedElastic())
			.cache());
}
```
this will reduce 3 operators and allocations